pkg/thanos/operator.go

```diff
@@ -739,10 +756,18 @@ func needsUpdate(pod *v1.Pod, tmpl v1.PodTemplateSpec) bool {
 	return false
 }
 
+func (o *Operator) enqueueForThanosRulerNamespace(nsName string) {
+	o.enqueueForNamespace(o.nsThanosRulerInf.GetStore(), nsName)
```

Comment:
What happens if this is called but we never call this line before:
> 		go o.nsThanosRulerInf.Run(stopc)

Reply:
In this case, `o.nsThanosRulerInf` is equal to `o.nsRuleInf` for which `Run()` is called.